internal/agent/http_cache/azureblob/getblob.go

```diff
@@ -76,14 +78,33 @@ func (azureBlob *AzureBlob) getBlob(writer http.ResponseWriter, request *http.Re
 	}
 
 	startProxyingAt := time.Now()
+	progressLoggedAt := startProxyingAt
 	// we usually proxy large files so let's use a larger buffer
 	largeBuffer := make([]byte, PROXY_DOWNLOAD_BUFFER_SIZE)
-	bytesRead, err := io.CopyBuffer(writer, resp.Body, largeBuffer)
-	if err != nil {
-		proxyingDuration := time.Since(startProxyingAt)
-		fail(writer, request, http.StatusInternalServerError, "failed to proxy cache entry download",
-			"err", err, "duration", proxyingDuration, "read", bytesRead, "key", key)
-
-		return
+	bytesRead := int64(0)
```

Comment:
What do you think about moving all of this into a separate function that we'll call instead of `io.CopyBuffer`?